lib/test.dart

```diff
@@ -41,6 +42,12 @@ export 'src/frontend/utils.dart';
 /// This is used if a test file is run directly, rather than through the runner.
 Declarer _globalDeclarer;
 
+typedef Reporter UserDefinedReporterCreateCallback(Engine engine);
+UserDefinedReporterCreateCallback _userDefinedReporterCallback;
+void setUserDefinedReporter(UserDefinedReporterCreateCallback callback) {
```

Comment:
Avoid methods named `setSomething`. You can have a `set` method at the top level, but since this isn't doing any extra work we can also make the variable public instead of private.

Reply:
@natebosch done